### DISCUSSION TITLE
Without mvvm page switching

### DISCUSSION THREAD

**Discussion by K0rkunc** [AvaloniaUI/Avalonia#10329] [Q&A]
Hello, how can I switch between pages without mvvm?

I could do it easily by doing push and pop in maui but I couldn't figure out how to do it in avalonia

Thanks a lot for your help.

**Comment by maxkatz6** [comment]
> Hello, how can I switch between pages without mvvm?

It's as basic as:

```xaml
<ContentControl x:Name="MyContentControl" />
```

```csharp
MyContentControl.Content = new Page1Control();
MyContentControl.Content = new Page2Control();
```

On top of which you can add more details. Like implementing a stack navigation or handling `TopLevel.GetTopLevel(view).BackRequested` (usable android/browser).

**Reply by K0rkunc** [reply]
Can you post a sample

**Comment by j4587698** [comment]
If you use ReactiveUI, you can use its routing system. I'm doing this.
